src/box/box.cc

```diff
@@ -1649,6 +1649,22 @@ box_check_wal_queue_max_size(void)
 	return size;
```

Comment:
The motivation here was:
> Since with the new behavior synchronous transactions can collect confirmations indefinitely, the queue can grow very large. You need to be able to limit the size of a synchronous queue. Therefore a new configuration option `replication_synchro_queue_max_size` has been added. This option puts a limit on a number of transactions in the master synchronous queue.

Therefore, it would be good to introduce an infinite timeout only after the queue has been limited.